src/rttest.cpp

```diff
@@ -281,7 +295,7 @@ int rttest_init_new_thread()
   auto thread_rttest_instance = get_rttest_thread_instance(thread_id);
   if (thread_rttest_instance == nullptr) {
     // Create the new Rttest instance for this thread
-    rttest_instance_map[thread_id] = new Rttest;
+    rttest_instance_map[thread_id] = new Rttest();
```

Comment:
Running the test and example with `valgrind leak-check=full` showed lost blocks due to this memory allocation. I deleted the thread instance in 3f85dbc in `finish()` and there are now no memory leaks.

There are, however, several "invalid read" errors ranging from size 1 to 8 (shown if valgrind is run with `-v`). I could spend some time trying to track these down.